Feat/api files notifications

Dartanyun opened the pull request:
Добавлены API ко всем оставшимся моделям, готов полностью рабочий API IDP. Удален ментор из юзера и перенесён в таск, исправлены фикстуры.

Dartanyun committed:
Create models.py

Here storing custom models

Dartanyun committed:
Few more models now have API

Created File, Notification, IDP, TaskNotification, IDPNotification base API

Dartanyun committed:
Model field migrations

Field mentor migrated from User model to Task model

Dartanyun committed:
FLAKE8 Fixes

Dartanyun committed:
little fix

Dartanyun committed:
Work

Dartanyun committed:
IDP Notification field fix

Dartanyun committed:
Merge branch 'dev' into feat/api_files_notifications

Dartanyun committed:
FLAKE8

makskhaliosa commented on api_v1/serializers/idp_app.py:
```diff
@@ -29,7 +63,61 @@ class Meta:
+            "end_date_plan",
+            "end_date_fact",
+            "employee",
+            "notifications",
```
Так как уже настроили автоматическое создание уведомлений, то можно не ожидать это поле, система найдет нужное уведомление по параметрам Ипр

Dartanyun replied:
@makskhaliosa не совсем понял коммент. Здесь в сериализаторе просто выводится инфа о уведомлениях и все.

makskhaliosa replied:
Надо было мне ниже написать. На чтение поле ок, а на создание, в create, можно не делать проверку на notifications, тк их не должно быть во входящих данных по идее.

Dartanyun replied:
@makskhaliosa там не проверка на наличие notifications, а возможность полноценно расписать эти самые notifications. Без них запись IDP не создается, но если что, могу отключить.

makskhaliosa replied:
Ок, пусть будет, просто отметил

makskhaliosa commented on api_v1/views/idp_app.py:
```diff
@@ -1,21 +1,62 @@
+    permission_classes = (AllowAny,)
+
+
+class FileViewSet(viewsets.ModelViewSet):
```
С файлами надо как-то еще продумать их сохранение при загрузке пользователем и их выгрузку с сервера, когда пользователь открывает задачу. Но это наверно отдельной задачей лучше сделать?

Dartanyun replied:
Да, отдельной задачей. За пример можно взять сохранение файлов как в киттиграме.

makskhaliosa commented on idp_app/admin.py:
```diff
@@ -45,35 +60,37 @@ class FileAdmin(admin.ModelAdmin):
     list_filter = ("status",)
     search_fields = ("name", "employee__last_name", "start_date")
     empty_value_display = "-пусто-"
+    inlines = [IDPNotificationTabularInline]
```
Клевая штука, можно для TaskAdmin тоже такое добавить с TaskNotification

Dartanyun replied:
Займусь

Dartanyun replied:
@makskhaliosa сделано

makskhaliosa reviewed (commented):
Все норм, оставил пару комментов

Rashid-creator-droid commented on users/migrations/0002_remove_user_mentor.py:
миграции

Dartanyun replied:
Уже имеются

Dartanyun committed:
Lil improvements

Now statuses choices in idp uses models.TextChoices. Mentor field added to IDP. Little improvements in admin panel

Dartanyun committed:
Auto status for IDP

Now if users' mentor create idp for him idp will be have active status, else status = draft

Dartanyun committed:
Fixtures update

Dartanyun committed:
improved readability

makskhaliosa commented on api_v1/views/idp_app.py:
```diff
@@ -87,3 +37,57 @@ class TaskViewSet(viewsets.ModelViewSet):
+        if user.is_authenticated:
+            idp_data = request.data
+
+            employee_cheif = User.objects.get(uid=idp_data["employee"]).chief
```
тут легко может упасть с Doesnotexist, лучше get_object_or_404()

Dartanyun replied:
mentor может быть null. Задаюсь вопросом может ли вообще ментор в IDP отсутствовать?

Dartanyun committed:
Merge branch 'dev' into feat/api_files_notifications

Dartanyun committed:
Task Inlines

Now TaskNotification model able to add new records in to Task model in admin panel

Dartanyun committed:
Merge branch 'feat/api_files_notifications' of https://github.com/makskhaliosa/idp_hacaton into feat/api_files_notifications

Rashid-creator-droid commented on core/choices.py:
```diff
@@ -11,6 +11,19 @@ class StatusChoices(models.TextChoices):
 
 
+class IdpStatuses(models.TextChoices):
+    """Status table for idps."""
```
докстринги на русском

Dartanyun replied:
Сделал

Rashid-creator-droid commented on api_v1/serializers/idp_app.py:
```diff
@@ -126,3 +116,34 @@ def create(self, validated_data):
         return idp
+
+
+class FileSerializer(serializers.ModelSerializer):
```
нет докстринга

Dartanyun replied:
Сделал

Dartanyun committed:
Fix

Dartanyun committed:
doc-strings

Dartanyun committed:
Merge branch 'dev' into feat/api_files_notifications

Dartanyun committed:
Fix
